src/Microsoft.Health.Fhir.SqlServer.UnitTests/Features/ChangeFeed/SqlServerFhirResourceChangeDataStoreTests.cs

```diff
@@ -75,5 +76,29 @@ public async Task GivenEmptyConnectionString_WhenGetResourceChanges_ThenInvalidO
                 Assert.Equal(nameof(InvalidOperationException), ex.GetType().Name);
             }
         }
+
+        [Fact]
+        public async Task GivenOperationCanceled_WhenGetResourceChanges_ThenNoExceptionThrown()
+        {
+            var mockConnection = new SqlConnectionStringBuilder
+            {
+                DataSource = "Test",
+                InitialCatalog = "Test",
+            }.ToString();
+
+            var config = Options.Create(new SqlServerDataStoreConfiguration { ConnectionString = mockConnection });
+            var connectionStringProvider = new DefaultSqlConnectionStringProvider(config);
+            var connectionFactory = new DefaultSqlConnectionFactory(connectionStringProvider);
+            var schemaInformation = new SchemaInformation(SchemaVersionConstants.Min, SchemaVersionConstants.Max);
+            schemaInformation.Current = SchemaVersionConstants.Max;
+            var resourceChangeDataStore = new SqlServerFhirResourceChangeDataStore(connectionFactory, NullLogger<SqlServerFhirResourceChangeDataStore>.Instance, schemaInformation);
```

Comment:
I think we need to re-use the constructor of the test class instead of redefining it from line 83 to line 94. If you need an actual test database call, you can add the test case to the integration test. Otherwise, you can adjust the constructor of the class and mock the resourceChangeDataStore.GetRecordsAsync function without making DB call.

Reply:
updated. had to make some small changes to the ctor since it was previously doing some string.Empty stuff for one of the tests which meant the code would hit an always exception there. but now it is configurable